app/v2/connectors/PersistenceConnector.scala

```diff
@@ -117,6 +122,27 @@ class PersistenceConnectorImpl @Inject() (httpClientV2: HttpClientV2, appConfig:
           }
     }
 
+  override def postMovementForLargeMessage(eori: EORINumber, movementType: MovementType)(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Future[MovementResponse] =
+    withMetricsTimerAsync(MetricsKeys.ValidatorBackend.Post) {
+      _ =>
+        val url = appConfig.movementsUrl.withPath(postMovementUrl(eori, movementType))
+
+        httpClientV2
+          .post(url"$url")
+          .execute[HttpResponse]
+          .flatMap {
+            response =>
+              response.status match {
+                case OK => response.as[MovementResponse]
+                case _ =>
+                  response.error
+              }
+          }
+    }
```

Comment:
We don't need to remove it since we already check for a None in contentTypeRouting

Reply:
Good point.

Reply:
yes, Initially started with the same approach to have an Option for Source -  if it's for small message we need to have something like  ```
 httpClientV2
          .post(url"$url")
          .transform(_.addHttpHeaders(HeaderNames.CONTENT_TYPE -> MimeTypes.XML))
          .withBody(source)
          .execute[HttpResponse] 
```
and for large message,    ```
httpClientV2
          .post(url"$url")
          .execute[HttpResponse]
```
instead of conditionally handling this, thought of better to separate method - so we can easily manage

Reply:
although, that said, I looked into the implementation of HMRC's withBody and it does a bit more than just set the body, so actually, yeah, we'd need to separate it out. 

So, something like

```scala
  override def postMovement(eori: EORINumber, movementType: MovementType, source: Option[Source[ByteString, _]])(implicit
    hc: HeaderCarrier,
    ec: ExecutionContext
  ): Future[MovementResponse] =
    withMetricsTimerAsync(MetricsKeys.ValidatorBackend.Post) {
      _ =>
        val url = appConfig.movementsUrl.withPath(postMovementUrl(eori, movementType))

        val httpClient = httpClientV2.post(url"$url")
        (source match {
          case Some(x) =>
            httpClient.transform(_.addHttpHeaders(HeaderNames.CONTENT_TYPE -> MimeTypes.XML)).withBody(x)
          case None => httpClient
        }).execute[HttpResponse]
          .flatMap {
            response =>
              response.status match {
                case OK => response.as[MovementResponse]
                case _ =>
                  response.error
              }
          }
    }
```

Reply:
```
  .execute[HttpResponse]
          .flatMap {
            response =>
              response.status match {
                case OK => response.as[MovementResponse]
                case _ =>
                  response.error
              }
          }
```

Can probably just be .execute[MovementResponse]?

Reply:
Yes, it could be.

Reply:
Updated the code to use Option[Source]